### ISSUE TITLE
Add support for DNS RRs, like HTTPS and SVCB

### ISSUE THREAD

**Issue by GamePad64** [Fyrd/caniuse#6091]
So, this is a new standard draft for DNS records, that is meant to be used in browsers. It is just like SRV, but a bit more advanced.

**Comment by Schweinepriester** [comment]
Got a link to the spec draft or something? :)

**Comment by Schweinepriester** [comment]
I mean I found <https://datatracker.ietf.org/doc/draft-ietf-dnsop-svcb-https/08/>. But only those two, type `HTTPS` as well as `SVCB`, or more?

<https://en.wikipedia.org/wiki/List_of_DNS_record_types> has them as well.

Also https://blog.cloudflare.com/speeding-up-https-and-http-3-negotiation-with-dns/ has some background, it appears.

**Comment by GamePad64** [comment]
Oh, sorry for having to look for spec by yourself. Yes, only these two record types are in the works now.
HTTPSSVC is just an older alias for HTTPS record.

FF tracking bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1634793
Chromium tracking bug: https://bugs.chromium.org/p/chromium/issues/detail?id=1052476

**Comment by GamePad64** [comment]
Also, shipped in Chrome 96: https://www.chromestatus.com/feature/5485544526053376

**Comment by PaperStrike** [comment]
HTTPS-RR consists of many parts, browsers implement them one by one, it may require lots of works sorting them out.

Chromium never seems to have a tracker for fully implementing HTTPS-RR, in addition to [#1052476 (archived for some reason)](https://bugs.chromium.org/p/chromium/issues/detail?id=1052476) and [#1206455 (HTTP->HTTPS redirect)](https://bugs.chromium.org/p/chromium/issues/detail?id=1206455), there are also [#1264933 (ECH support)](https://bugs.chromium.org/p/chromium/issues/detail?id=1264933) and [#1417033 (AliasMode support and ServiceMode target support)](https://bugs.chromium.org/p/chromium/issues/detail?id=1417033).

The last one was just opened a day ago, but I personally think it's the most wanted feature.

**Comment by jschauma** [comment]
FYI, the draft is now an RFC https://www.rfc-editor.org/rfc/rfc9460.html

I did some research on the adoption of the HTTPS record:
https://www.netmeister.org/blog/https-rrs.html

Relevant findings here:
Safari seems to support it fully; Firefox supports it fully iff using DoH; Chrome currently only supports the 'ech' field.

Having these records and support for all fields tracked in caiuse.com would be very useful indeed!

**Comment by otbutz** [comment]
> Firefox supports it fully if using DoH

Also for system DNS since Firefox 129: https://www.mozilla.org/en-US/firefox/129.0/releasenotes/

**Comment by jschauma** [comment]
Not on macOS, though.  (The release notes say that, but of course I did just waste a few minutes wondering wtf it wasn't working. ;- )

**Comment by 0x192** [comment]
@jschauma 

Did you enable `network.dns.native_https_query` in `about:config` ? On Linux, I had to do this.

**Comment by jschauma** [comment]
For what it's worth, I put a mapping of the most important features to browser support here: https://www.netmeister.org/blog/https-caniuse.html

In a nutshell, as far as I've observed today:
- All browsers support ALPN
- Safari has the best support; it's the only browser that supports AliasMode, but Safari doesn't support ECH
- Firefox requires DoH, but then supports several parameters
- Chrome only support ALPN and ECH

(There are many additional edge cases to test, but those are the main things I often care about.)

**Comment by Gunni** [comment]
`no-default-alpn` needs to be listed as well.

**Comment by jschauma** [comment]
Yes, just added that.  Looks to me like none of the browsers DTRT here; all fail over (or race) connections not specified.

**Comment by IzumiSenaSora** [comment]
> For what it's worth, I put a mapping of the most important features to browser support here: https://www.netmeister.org/blog/https-caniuse.html

Is this up-to-date?

**Comment by jschauma** [comment]
> > For what it's worth, I put a mapping of the most important features to browser support here: https://www.netmeister.org/blog/https-caniuse.html
> 
> Is this up-to-date?

It reflects what I saw as of 2025-12-12.

**Comment by chrysn** [comment]
It is not up to date any more: Firefox's support has gotten worse as it
regressed around port handling.

]https://bugzilla.mozilla.org/show_bug.cgi?id=2023355

**Comment by jschauma** [comment]
FWIW, not saying there wasn't a regression, but I did not test websockets in my analysis and only looked at boring old https requests made by the browser for a simple static site without websockets.
